src/lib/ndpi_main.c

```diff
@@ -7768,9 +7765,20 @@ ndpi_protocol ndpi_guess_undetected_protocol(struct ndpi_detection_module_struct
       }
     }
 
-    if(ndpi_search_into_bittorrent_cache(ndpi_str, NULL /* flow */,
-					 htonl(shost), htons(sport),
-					 htonl(dhost), htons(dport))) {
+    /* Workaround: create a fake flow struct since ndpi_search_into_bittorrent_cache is ipv4/6 agnostic,
+       while this function is intrinsecally ipv4 only.
+       This way we don't need to change the (public) ndpi_guess_undetected_protocol() prototype */
```

Comment:
Why not changing the public api? We're doing that soon often. Why not this time? The result would be cleaner code. :)

Reply:
I will talk with @lucaderi about that!